src/language/parser.py

```diff
@@ -162,9 +152,8 @@ def parse_yaml_rules(self, nodelist, base_class=None):
             else:
                 args.base_class = base_class if base_class else 'Ast'
 
-                # check if we need token for the node
-                # todo : we will have token for every node
-                args.has_token = LanguageParser.is_token(class_name)
+                # store token in every node
+                args.has_token = True
```

Comment:
Remove has_token from everywhere and simply add token to every node

Reply:
It is fine like this at this point, as only the base classes do not get a token member. These classes are not abstract, so they can be instantiated by the user. In the future, a token member can be added to them to be able to be used by the user.